### PR TITLE
feat: 🎸 Only support static types at hosts

### PR DESCRIPTION
**Pull request by calcaide**
Make sure the hosts list within a host-catalog just show rows that have a Static type.

[Direct link](https://boundary-ui-git-feature-host-catalog-hosts-onl-f4d92e-hashicorp.vercel.app/scopes/s_xk1i4dk50v2/host-catalogs/1/hosts).

Screen Shot 2021-12-01 at 4 03 41 PM

### CODE REVIEW

File: addons/api/addon/models/host.js [hunk 1]
```diff
@@ -7,4 +7,11 @@ export default class HostModel extends GeneratedHostModel {
    * @type {FragmentHostAttributesModel}
    */
   @fragment('fragment-host-attributes', { defaultValue: {} }) attributes;
+
+  /**
+   * @type {boolean}
+   */
+  get isStaticType() {
```
Comment (randallmorey) [inline addons/api/addon/models/host.js:14]: `isStatic` ought to do it and is more consistent with existing field names.

File: addons/api/tests/unit/models/host-test.js [hunk 2]
```diff
@@ -20,4 +22,14 @@ module('Unit | Model | host', function (hooks) {
     });
     assert.equal(model.attributes.address, '127.0.0.1');
   });
+
+  test('it has isStaticType function and returns the expected values', async function (assert) {
+    assert.expect(3);
+    const store = this.owner.lookup('service:store');
+    const model = store.createRecord('host', { type: 'static' });
```
Comment (randallmorey) [inline addons/api/tests/unit/models/host-test.js:29]: how about `modelA` for consistency?

**Review by randallmorey — CHANGES_REQUESTED** [review]

**Review by randallmorey — CHANGES_REQUESTED** [review]
Sorry for the confusion.  The host catalog index (not the host index) should show only static host catalogs in the UI for now.

**Comment by calcaide** [comment]
@randallmorey 

Ok. So we need to add the `type: static` constrain to host-catalogs list ([link here](https://boundary-ui-git-feature-host-catalog-hosts-onl-f4d92e-hashicorp.vercel.app/scopes/s_xk1i4dk50v2/host-catalogs)) and do not add any constrain on the hosts list within a host-catalog ([link here](https://boundary-ui-git-feature-host-catalog-hosts-onl-f4d92e-hashicorp.vercel.app/scopes/s_xk1i4dk50v2/host-catalogs/1/hosts)).

Is this correct?

**Comment by calcaide** [comment]
As per comments on this PR and [this slack thread](https://hashicorp.slack.com/archives/C012GTH1C4E/p1638827922072500) closing this PR to move the support static type hosts to host-catalogs index.

**Pull request closed without merging**